src/openassetio-core/tests/trait/TraitBaseTest.cpp

```diff
@@ -56,21 +56,40 @@ SCENARIO("Retrieving the undelying data") {
 }
 
 SCENARIO("Checking a trait is imbued") {
-  GIVEN("a trait view on an empty TraitsData") {
+  GIVEN("an empty TraitsData") {
     const TraitsDataPtr data = TraitsData::make();
-    const TestTrait trait(data);
 
-    WHEN("the view is queried for the imbued status") {
-      const bool isImbued = trait.isImbued();
+    AND_GIVEN("an instance of a trait view wrapping the TraitsData") {
+      const TestTrait trait(data);
+
+      WHEN("the view instance is queried for the imbued status") {
+        const bool isImbued = trait.isImbued();
+
+        THEN("the view reports that the trait is not imbued") { CHECK(!isImbued); }
+      }
+
+      AND_GIVEN("the TraitsData is imbued with the view's trait") {
+        data->addTrait(TestTrait::kId);
+
```

Comment:
There is at least now internal consistency in each of `TraitBase` and `TraitsData`. 

They're not wholly consistent with one-another, it's true. 

If we changed `TraitsData` to use `imbue` rather than `addTrait`, and `isImbued` rather than `hasTrait`, then does that have a knock-on effect for e.g. `get`/`setTraitProperty`?

I think `TraitBase` representing a singular trait and being a higher level interface compared to the the low-level `TraitsData` collection (which ideally is just an opaque blob to users) kinda justifies a different nomenclature.

Reply:
This would break symmetry with the other `TraitsData` methods though. Right now, 'imbued' is a concept that exists solely in the 'view' classes.  We could tackle renaming the entire TraitsData API as a separate topic, but its consistent as stands at least and we'd need a concrete benefit. IIRC we used 'imbue' as the more normal wordage was confusing (for some reason I forget) when it comes to the view classes, in a way that wasn't an issue with the lower-level data container.

Reply:
@feltech I think we misunderstood this. I think @elliotcmorris was saying should the code be using the method on `TraitBase` not manipulating the data. My take on that was that the reason we manipulate the data directly in a `GIVEN` is that we know then the starting state is correct. Using a `TraitBase` method could (with admitedly low probability) combine two bugs that meant we aren't actually guaranteeing the behaviour is as expected.